libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -231,6 +232,45 @@ bool StaticAnalyzer::visit(InlineAssembly const& _inlineAssembly)
 	return true;
 }
 
+bool StaticAnalyzer::visit(BinaryOperation const& _operation)
+{
+	if (
+		_operation.rightExpression().annotation().isConstant &&
+		(_operation.getOperator() == Token::Div || _operation.getOperator() == Token::Mod)
+	)
+		if (auto result = dynamic_pointer_cast<RationalNumberType const>(
+			ConstantEvaluator(m_errorReporter).evaluate(_operation.rightExpression())
+		))
+			if (result->literalValue(nullptr) == u256(0))
```

Comment:
Will this not categorise `0/1` as failure ?

Reply:
Only the right hand side is evaluated - the variable name ``result`` was chosen a bit poorly - I'll change it to ``rhs``.